internal/kafka/internal/services/quota/internal/utils/simple_billing_model_resolver.go

```diff
@@ -0,0 +1,145 @@
+package utils
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/errors"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/shared/utils/arrays"
+	amsv1 "github.com/openshift-online/ocm-sdk-go/accountsmgmt/v1"
+)
+
+/********************************************************************************************************************
+ * This resolver is used when the user didn't specify the `billing model` and the `marketplace` in the kafka request.
+ * The billing model is resolved this way: if the user has quota for `standard`, it returns `standard`. If he has
+ * quota for `marketplace` it returns `marketplace`.
+ *
+ * Attempts are done to resolve the marketplace type too.
+ ********************************************************************************************************************/
+
+var _ BillingModelResolver = &simpleBillingModelResolver{}
+
+type simpleBillingModelResolver struct {
+	quotaConfigProvider AMSQuotaConfigProvider
+	kafkaConfig         *config.KafkaConfig
+}
+
+func (*simpleBillingModelResolver) SupportRequest(kafka *dbapi.KafkaRequest) bool {
+	return kafka.BillingCloudAccountId == "" && kafka.Marketplace == ""
+}
+
+// Resolve - tries to resolve the KafkaBillingModel and amsBillingModel by analysing the parameter received by the user.
+// This function checks if for the requested instance type the `standard` and `marketplace` billing model are defined
+// and produces a sorted array contained those billing models (if defined) in the priority order (standard first,
+// marketplace second), then demands the resolving logic to the `resolve` method.
+func (r *simpleBillingModelResolver) Resolve(orgId string, kafka *dbapi.KafkaRequest) (BillingModelDetails, error) {
+	// try standard and marketplace
+	var kafkaBillingModels []config.KafkaBillingModel
+	standardBillingModel, err := r.kafkaConfig.GetBillingModelByID(kafka.InstanceType, "standard")
+	if err == nil {
+		kafkaBillingModels = append(kafkaBillingModels, standardBillingModel)
+	}
+	marketplaceBillingModel, err := r.kafkaConfig.GetBillingModelByID(kafka.InstanceType, "marketplace")
+	if err == nil {
+		kafkaBillingModels = append(kafkaBillingModels, marketplaceBillingModel)
+	}
+	trialBillingModel, err := r.kafkaConfig.GetBillingModelByID(kafka.InstanceType, "trial")
```

Comment:
This is referencing `trial`. Is that correct after recent changes where trial supported billing model does not exist anymore?

Reply:
Yeah. We have to remove it.